dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SetShippingMethodsOnCartTest.php

```diff
@@ -42,100 +48,437 @@ protected function setUp()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'test_order_with_virtual_product_without_address';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = $this->getQuoteAddressIdByReversedQuoteId($reservedOrderId);
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        self::expectException(\Exception::class);
+        self::expectExceptionMessage('Carrier with such method not found: ' . $methodCode . ', ' . $methodCode . '');
+        $this->graphQlQuery($query);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_simple_product_saved.php
+     */
     public function testShippingMethodWithSimpleProduct()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'test_order_with_simple_product_without_address';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = $this->getQuoteAddressIdByReversedQuoteId($reservedOrderId);
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        $response = $this->graphQlQuery($query);
+
+        self::assertArrayHasKey('setShippingMethodsOnCart', $response);
+        self::assertArrayHasKey('cart', $response['setShippingMethodsOnCart']);
+        self::assertArrayHasKey('shipping_addresses', $response['setShippingMethodsOnCart']['cart']);
+
+        $shippingMethod = current($response['setShippingMethodsOnCart']['cart']['shipping_addresses']);
+        self::assertArrayHasKey('selected_shipping_method', $shippingMethod);
+        self::assertEquals($carrierCode, $shippingMethod['selected_shipping_method']['carrier_code']);
+        self::assertEquals($methodCode, $shippingMethod['selected_shipping_method']['method_code']);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_simple_product_saved.php
+     * @expectedException \Exception
+     * @expectedExceptionMessage Required parameter "cart_address_id" is missing.
+     */
     public function testShippingMethodWithSimpleProductWithoutAddress()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'test_order_with_simple_product_without_address';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = 0;
```

Comment:
Logic error
We passed address_id but we expected `Required parameter "cart_address_id" is missing.`
So need to fix logic in implementation

Example of Required parameters testcase you could find in:
https://github.com/magento/graphql-ce/blob/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SetShippingAddressOnCartTest.php#L223

Reply:
Modified test case as per given reference link, please review it.